test/openai/test_completion.py

```diff
@@ -140,8 +140,373 @@ def success_metrics(responses, prompt, test, entry_point):
+        - units
+        - fractions
+        - square roots
+        - superfluous LaTeX.
+        Source: https://github.com/hendrycks/math
+        """
+        if str1 is None and str2 is None:
+            print("WARNING: Both None")
+            return 1.0
+        if str1 is None or str2 is None:
+            return 0.0
+
+        try:
+            ss1 = _strip_string(str1)
+            ss2 = _strip_string(str2)
+            return float(ss1 == ss2)
+        except Exception:
+            return float(str1 == str2)
+
+    def is_equiv_chain_of_thought(str1: str, str2: str) -> float:
+        """Strips the solution first before calling `is_equiv`."""
+        ans1 = get_answer(str1)
+        ans2 = get_answer(str2)
+
+        return is_equiv(ans1, ans2)
+
+    def success_metrics(responses, solution, **args):
+        """Check if each response is correct.
+
+        Args:
+            responses (list): The list of responses.
+            solution (str): The canonical solution.
+
+        Returns:
+            dict: The success metrics.
+        """
+        success_list = []
+        n = len(responses)
+        for i in range(n):
+            response = responses[i]
+            succeed = is_equiv_chain_of_thought(response, solution)
+            success_list.append(succeed)
+        return {
+            "expected_success": 1 - pow(1 - sum(success_list) / n, n),
+            "success": any(s for s in success_list),
+        }
+
+    seed = 41
+    data = datasets.load_dataset("competition_math")
+    train_data = data["train"].shuffle(seed=seed)
+    test_data = data["test"].shuffle(seed=seed)
+    n_tune_data = 20
+    tune_data = [
+        {
+            "problem": train_data[x]["problem"],
+            "solution": train_data[x]["solution"],
+        }
+        for x in range(len(train_data))
+        if train_data[x]["level"] == "Level 1"
+    ][:n_tune_data]
+    test_data = [
+        {
+            "problem": test_data[x]["problem"],
+            "solution": test_data[x]["solution"],
+        }
+        for x in range(len(test_data))
+        if test_data[x]["level"] == "Level 1"
+    ]
+    print(
+        "max tokens in tuning data's canonical solutions",
+        max([len(x["solution"].split()) for x in tune_data]),
+    )
+    print(len(tune_data), len(test_data))
+    # prompt template
+    prompts = [
+        lambda data: "Given a mathematics problem, determine the answer. Simplify your answer as much as possible.\n###\nProblem: What is the value of $\\sqrt{3! \\cdot 3!}$ expressed as a positive integer?\nAnswer: $\\sqrt{3!\\cdot3!}$ is equal to $\\sqrt{(3!)^2}=3!=3\\cdot2\\cdot1=\\boxed{6}$.\n###\nProblem: %s\nAnswer:"
+        + data["problem"]
+    ]
+
+    try:
+        oai.ChatCompletion.set_cache(seed)
+        vanilla_config = {
+            "model": "gpt-3.5-turbo",
+            "temperature": 1,
+            "max_tokens": 2048,
```

Comment:
What if we remove this?

Reply:
It is also fine to remove if preferred. It does not give any error.

Reply:
What is the intention for `vanilla_config`? If the intention is the default config, then it should be removed.